bd sync crashes with nil pointer dereference during JSONL export (v0.9.11)

## Environment

- **bd version:** 0.9.11 (dev)
- **OS:** macOS 15.6.1 (Sequoia)
- **Architecture:** arm64 (Apple Silicon)
- **Installation:** Homebrew (`/opt/homebrew/bin/bd`)
- **Binary type:** Mach-O 64-bit executable

## Steps to Reproduce

1. Initialize beads in a git repository:
   ```bash
   bd init
   ```

2. Create some issues:
   ```bash
   bd create "Test issue"
   ```

3. Attempt to sync manually:
   ```bash
   bd sync
   ```

## Expected Behavior

The command should complete the sync workflow:
1. Export pending changes to JSONL
2. Commit changes to git
3. Pull from remote
4. Import updated JSONL
5. Push local commits to remote

(As shown in dry-run mode: `bd sync --dry-run` completes successfully)

## Actual Behavior

The command crashes immediately during the export phase with a nil pointer dereference:

```
→ Exporting pending changes to JSONL...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x118 pc=0x101144da8]

goroutine 1 [running]:
main.exportToJSONL({0x1013f6820, 0x101950680}, {0x1400015ca20, 0x2f})
	github.com/steveyegge/beads/cmd/bd/sync.go:245 +0x48
main.init.func40(0x1019217a0, {0x1011895d0?, 0x4?, 0x1011894f8?})
	github.com/steveyegge/beads/cmd/bd/sync.go:73 +0x39c
github.com/spf13/cobra.(*Command).execute(0x1019217a0, {0x101950680, 0x0, 0x0})
	github.com/spf13/cobra@v1.10.1/command.go:1019 +0x7bc
github.com/spf13/cobra.(*Command).ExecuteC(0x10191e100)
	github.com/spf13/cobra@v1.10.1/command.go:1148 +0x350
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.10.1/command.go:1071
main.main()
	github.com/steveyegge/beads/cmd/bd/main.go:2110 +0x24
```

## Additional Context

### ✅ What Works:
- `bd export -o /path/to/file.jsonl` - Exports successfully
- `bd sync --dry-run` - Completes without error
- `bd daemon` automatic sync cycles - Work perfectly (daemon logs show successful exports)
- All database operations and issue management commands

### ❌ What Crashes:
- `bd sync` (manual CLI invocation)

### Analysis:

The crash location (`sync.go:245` in `exportToJSONL`) combined with the memory address `0x118` (280 bytes offset) suggests a struct field that's expected to be initialized but isn't when `sync` is invoked from the CLI.

The daemon's sync implementation appears to use a different code path or initialization sequence that properly sets up all required context before calling the export function.

**Potential root cause:** The `sync` command at line 73 passes parameters to `exportToJSONL`, but one of those parameters (likely a configuration struct or database connection context) is nil when called from the CLI but properly initialized when called from the daemon.

## Database State

Database is properly configured with all required tables:
```
issues, dependencies, labels, comments, sqlite_sequence, events, 
config, metadata, dirty_issues, issue_counters, issue_snapshots, 
compaction_snapshots
```

Configuration appears correct:
```
bd_version: 0.9.11
issue_prefix: spacewalker
(all other config values properly set)
```

## Impact

- **Severity:** Medium - Workaround available (use daemon or manual git commits)
- **User Impact:** Cannot use `/beads:sync` slash command or manual `bd sync` CLI
- **Data Safety:** Can cause data loss if crash occurs after truncating JSONL file but before writing new data

## Workarounds

1. Use the daemon for automatic syncing (recommended):
   ```bash
   bd daemon --auto-commit --auto-push
   ```

2. Manual git workflow:
   ```bash
   bd export -o .beads/issues.jsonl
   git add .beads/issues.jsonl
   git commit -m "chore: update beads issues"
   git push
   ```

3. Avoid using the `/beads:sync` slash command in Claude Code

## Suggested Fix

The issue likely requires ensuring that the sync command properly initializes all context (database connection, configuration, etc.) before calling `exportToJSONL`. The daemon's initialization sequence could serve as a reference for the correct setup.

--- a/cmd/bd/sync.go
+++ b/cmd/bd/sync.go
@@ -258,6 +258,11 @@ func exportToJSONL(ctx context.Context, jsonlPath string) error {
 	}
 
 	// Direct mode: access store directly
+	// Ensure store is initialized
+	if err := ensureStoreActive(); err != nil {
+		return fmt.Errorf("failed to initialize store: %w", err)
+	}
+
 	// Get all issues
 	issues, err := store.SearchIssues(ctx, "", types.IssueFilter{})
 	if err != nil {

